src/react/components/common/assetPreview/videoAsset.tsx

```diff
@@ -225,4 +302,66 @@ export class VideoAsset extends React.Component<IVideoAssetProps> {
         }
     }
+
+    /**
+     * @name - Add Asset Timeline Tags
+     * @description - Draws small lines to show where visited and tagged frames are on
+     * the video line
+     * @member childAssets - Array of child assets in the video
+     * @member videoDuration - Length (in seconds) of the video
+     */
+    private addAssetTimelineTags(childAssets: any[], videoDuration: number) {
+        const innerHtml: string = this.getRenderedAssetTagLinesText(childAssets, videoDuration);
+        if (!this.timelineElement) {
+            // Find the progress holder so we can add the markers to that
+            const progressControls = document.getElementsByClassName("video-react-progress-control");
+            let progressHolderElement: Element = null;
+
+            // Find any that have an editor page content body type, this will be the one we use
+            Array.from(progressControls).forEach( (element) => {
+                let testingElement: Element = element;
+                while (testingElement) {
+                    if (testingElement.className === "editor-page-content-body") {
+                        progressHolderElement = element;
+                    }
+
+                    testingElement = testingElement.parentElement;
+                }
+            });
+
+            // If we found an element to hold the tags, add them to it
+            if (progressHolderElement) {
+                this.timelineElement = document.createElement("div");
+                this.timelineElement.className = "video-timeline-parent";
+                this.timelineElement.innerHTML = innerHtml;
+                progressHolderElement.appendChild(this.timelineElement);
+            }
+        } else {
+            this.timelineElement.innerHTML = innerHtml;
+        }
+    }
+
+    /**
+     * @name - Get Rendered Asset Tag Lines Text
+     * @description - Gets the HTML text for the rendered asset tag lines
+     * @member childAssets - Array of child assets in the video
+     * @member videoDuration - Length (in seconds) of the video
+     */
+    private getRenderedAssetTagLinesText(childAssets: any[], videoDuration: number): string {
+        const tagTimeLines: any = [];
+
+        // Add some markers for frames that have been visited with yellow and tagged with green
+        for (const childAsset of childAssets) {
+            // Calcualte the left position
+            const childPosition: number = (childAsset.timestamp / videoDuration);
+            tagTimeLines.push(<div key={childAsset.timestamp}
+                className = {childAsset.state === AssetState.Tagged ?
+                    "video-timeline-tagged" : "video-timeline-untagged"}
+                style={{
+                    left: (childPosition * 100) + "%",
+                 }} />);
+        }
+        const taggedAssetDiv = <div>{tagTimeLines}</div>;
+        return ReactDOMServer.renderToStaticMarkup(taggedAssetDiv);
```

Comment:
I've never used this `ReactDOMServer` before... how is it performing?  I'm curious what is the different between this and `ReacDOM.render(...)`.  If you take a look @ `/src/index.tsx` the root of the app is found and rendered with the later approach.  I'm thinking this may be closer to the recommended way of doing what you are attempting here - Creating your root element, then rendering in a bunch of components inside of it?

Reply:
I originally used ReactDOM.Render, but that seemed to simply replace the children rather than append them, but it's possible I was misunderstanding what "container" meant in that context, I will look into it.

Reply:
Updated, I had to switch no matter what, because the static markup wouldn't work with onClick events.